app/components/channel_item/channel_item.tsx

```diff
@@ -0,0 +1,265 @@
+
+import CustomStatus from './custom_status';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+
+type Props = {
+    channel: ChannelModel;
+    collapsed: boolean;
+    currentUserId: string;
+    hasDraft: boolean;
+    isActive: boolean;
+    isInfo?: boolean;
+    isMuted: boolean;
+    isVisible: boolean;
+    myChannel?: MyChannelModel;
+    onPress?: (channelId: string) => void;
+    teamDisplayName?: string;
+    testID?: string;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        flexDirection: 'row',
+        paddingHorizontal: 20,
+        minHeight: 40,
+        alignItems: 'center',
+    },
+    infoItem: {
+        paddingHorizontal: 0,
+    },
+    wrapper: {
+        flex: 1,
+        flexDirection: 'row',
+    },
+    icon: {
+        fontSize: 24,
+        lineHeight: 28,
+        color: changeOpacity(theme.sidebarText, 0.72),
+    },
+    text: {
+        marginTop: -1,
+        color: changeOpacity(theme.sidebarText, 0.72),
+        paddingLeft: 12,
+    },
+    highlight: {
+        color: theme.sidebarText,
+    },
+    textInfo: {
+        color: theme.centerChannelColor,
+        paddingRight: 20,
+    },
+    muted: {
+        color: changeOpacity(theme.sidebarText, 0.4),
+    },
+    badge: {
+        position: 'relative',
+        left: 0,
+        top: 0,
+        alignSelf: undefined,
+    },
+    infoBadge: {
+        color: theme.buttonColor,
+        backgroundColor: theme.buttonBg,
+        borderColor: theme.centerChannelBg,
+    },
+    mutedBadge: {
+        opacity: 0.4,
+    },
+    activeItem: {
+        backgroundColor: changeOpacity(theme.sidebarTextActiveColor, 0.1),
+        borderLeftColor: theme.sidebarTextActiveBorder,
+        borderLeftWidth: 5,
+        marginLeft: 0,
+        paddingLeft: 14,
+    },
+    textActive: {
+        color: theme.sidebarText,
+    },
+    teamName: {
+        color: changeOpacity(theme.centerChannelColor, 0.64),
+        paddingLeft: 12,
+        marginTop: 4,
+        ...typography('Body', 75),
+    },
+    teamNameTablet: {
+        marginLeft: -12,
+        paddingLeft: 0,
+        marginTop: 0,
+        paddingBottom: 0,
+        top: 5,
+    },
+}));
+
+const textStyle = StyleSheet.create({
+    bright: typography('Body', 200, 'SemiBold'),
+    regular: typography('Body', 200, 'Regular'),
+});
+
+const ChannelListItem = ({
+    channel, collapsed, currentUserId, hasDraft,
+    isActive, isInfo, isMuted, isVisible,
+    myChannel, onPress, teamDisplayName, testID}: Props) => {
+    const {formatMessage} = useIntl();
+    const theme = useTheme();
+    const isTablet = useIsTablet();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const teammateId = (channel.type === General.DM_CHANNEL) ? getUserIdFromChannelName(currentUserId, channel.name) : undefined;
```

Comment:
Apparently, we can move this below the `return null`, so we don't have to do these calculations if we don't render the component.